### PR TITLE
fix and un-ignore concurrent key mutations tests

### PR DESCRIPTION
**Pull request by madadam**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @maqi

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: tests/maid_manager.rs [hunk 1]
```diff
@@ -497,139 +497,94 @@ fn account_balance_with_failed_mutations_with_churn() {
...
+                assert!(!new_keys.contains(&key));
             }
         }
 
-        match client.list_auth_keys_and_version_response(&mut nodes) {
-            Ok(result) => assert_eq!(result, (auth_keys.clone(), version)),
-            Err(err) => panic!("Unexpected error {:?} when list auth_keys and version", err),
-        }
+        stored_keys = new_keys.into_iter().collect();
+        version = new_version;
```
Comment (maqi) [inline tests/maid_manager.rs]: maybe we shall also confirm that:
asser_eq(new_version , version  + (ins_successes.len() + del_successes.len()) )
?

File: tests/maid_manager.rs [hunk 2]
```diff
@@ -497,139 +497,94 @@ fn account_balance_with_failed_mutations_with_churn() {
...
-            (true, true) => version += 2,
-            (true, false) => {
-                version += 1;
-                let _ = auth_keys.insert(app_key);
-            }
-            // TODO: advance clock and create another even to trigger expiration check.
-            (false, false) => panic!("Insertion is expected to always succeed"),
-            (false, true) => panic!("Deletion shall not be succeed when insertion is failed"),
-        }
+        assert!(ins_successes.len() + del_successes.len() <= 1,
```
Comment (maqi) [inline tests/maid_manager.rs]: there is chance that after one conflicting operation, the account is out-of-sync.
shall we put it as a TODO comment of limitation?

**Comment by madadam** [comment]
Superseded by https://github.com/maidsafe/safe_vault/pull/625. Closing.

**Pull request closed without merging**